### DISCUSSION TITLE
implement Dialog.Positioner

### DISCUSSION THREAD

**Discussion by Newbie012** [radix-ui/primitives#3934] [Ideas]
### Overview

Currently, implementing dialogs is quite intuitive. Although, when it comes to enter/exit animations, the styling of `DialogContent` becomes a bit cumbersome. I would like to suggest an option to pass `Dialog.Positioner` which allows easier implementation of animation presence.

```diff
import * as Dialog from '@radix-ui/react-dialog';

export default () => (
  <Dialog.Root>
    <Dialog.Trigger />
    <Dialog.Portal>
      <Dialog.Overlay />
+       <Dialog.Positioner>
          <Dialog.Content>
            <Dialog.Title />
            <Dialog.Description />
            <Dialog.Close />
          </Dialog.Content>
+       <Dialog.Positioner>
    </Dialog.Portal>
  </Dialog.Root>
);
```

### Examples in other libraries

Ark UI's [dialog primitive](https://ark-ui.com/docs/components/dialog) is almost identical to this library's Dialog. Their `Positioner` implementation looks great.

### Who does this impact? Who is this for?

Anyone who wants a simplified way of animating dialog presence without tweaking transform origin / odd translation percentage.

**Comment by chaance** [comment]
No plans to implement at this time.
